src/InstrumentationEngine/Instruction.cpp

```diff
@@ -1297,41 +1126,31 @@ HRESULT MicrosoftInstrumentationEngine::CSwitchInstruction::SetBranchTarget(_In_
     HRESULT hr = S_OK;
     CLogging::LogMessage(_T("Starting CSwitchInstruction::SetBranchTarget"));
 
-    m_branchTargets[index] = pTarget;
-
-    CLogging::LogMessage(_T("End CSwitchInstruction::SetBranchTarget"));
+    CComPtr<CInstruction> oldTarget = m_branchTargets[index];
+    m_branchTargets[index] = (CInstruction*)(pTarget);
+    CBranchTargetInfo::SetBranchTarget(this, (CInstruction*)(pTarget), oldTarget);
```

Comment:
Did you want to propagate the hresult here?

Reply:
Yup, fixed.